gap_fit_module.f95

```diff
@@ -254,37 +320,41 @@ subroutine gap_fit_parse_command_line(this)
           If >>T<<, they are interpreted as per-atom errors, and the variance will be scaled according to the number of atoms in the configuration. &
           If >>F<< they are treated as absolute errors and no scaling is performed. &
           NOTE: values specified on a per-configuration basis (see >>kernel_regularisation_parameter_name<<) are always absolute, not per-atom.", altkey="sigma_per_atom")
-   
+
      call param_register(params, 'do_copy_atoms_file', 'T', do_copy_at_file, &
           help_string="Copy the input XYZ file into the GAP XML file (should be set to False for NetCDF input).", altkey="do_copy_at_file")
-   
+
      call param_register(params, 'sparse_separate_file', 'T', sparseX_separate_file, &
           help_string="Save sparse point data in separate file in binary (use it for large datasets)")
-   
+
      call param_register(params, 'sparse_use_actual_gpcov', 'F', sparse_use_actual_gpcov, &
           help_string="Use actual GP covariance for sparsification methods")
-   
+
      call param_register(params, 'gap_file', 'gap_new.xml', gp_file, &
           help_string="Name of output XML file that will contain the fitted potential", altkey="gp_file")
-   
+
      call param_register(params, 'verbosity', 'NORMAL', verbosity, &
           help_string="Verbosity control. Options: NORMAL, VERBOSE, NERD, ANALYSIS.") ! changed name to ANALYSIS now that we are grown up
-   
+
      call param_register(params, "rnd_seed", "-1", rnd_seed, &
           help_string="Random seed.")
-   
+
      call param_register(params, "openmp_chunk_size", "1", openmp_chunk_size, &
           help_string="Chunk size in OpenMP scheduling")
-   
+
      call param_register(params, 'do_ip_timing', 'F', do_ip_timing, &
           help_string="To enable or not timing of the interatomic potential.")
-   
+
      call param_register(params, 'template_file', 'template.xyz', template_file, has_value_target=this%has_template_file, &
           help_string="Template XYZ file for initialising object")
 
      call param_register(params, 'sparsify_only_no_fit', 'F', sparsify_only_no_fit, &
           help_string="If true, sparsification is done, but no fitting. print the sparse index by adding print_sparse_index=file.dat to the descriptor string.")
-     
+
+     call param_register(params, 'do_serial', 'F', do_serial, help_string="Disable MPI, ScaLAPACK, and task manager")
```

Comment:
We should discuss the best name for this option, since `do_serial=T` still implies OpenMP parallelisation.

Reply:
I am open to better names.

I just realised that this option is not needed right now since I readded the condition "n_procs > 1" for ScaLAPACK after debugging. It still does disable the MPI_obj before initialising the ScaLAPACK_obj and the task_manager.

Reply:
Removed. Is now outdated.